### DISCUSSION TITLE
offlineExchange failing to save to IndexedDB due to variables proxy

### DISCUSSION THREAD

**Discussion by alexjharrison** [urql-graphql/urql#1809] [Q&A] [answered]
Hi all, thanks for the work you put into this awesome library.

I've come across a weird bug when I'm offline and trying to execute a mutation.  The offline exchange is sucessfully catching the operation and saving it to the failed queue.  

The problem comes when it tries to save the operation to the IndexedDB storage here.  
https://github.com/FormidableLabs/urql/blob/4396469385ae61abc0853ee9e1691ab5d18e4ead/exchanges/graphcache/src/offlineExchange.ts#L85-L97

I get this error

![image](https://user-images.githubusercontent.com/8920897/125995112-882fac4e-4c62-4fff-ad28-3a4ebb3a4e90.png)

operation.variables is a Proxy wrapped object in @urql/vue.  I read that proxies can't be saved into IDB.  So I went into the code and wrapped the results with 
```javascript
JSON.parse(JSON.strigify(results))
```
and it seems to have saved successfully.

Let me know if I'm looking in the wrong spot,
Thanks

**Comment by JoviDeCroock** [comment, marked as answer]
I think that sounds like a correct approach, we could check if it's a proxy and execute `JSON.parse(JSON.strigify(operation.variables))` however we could also opt to send it as a non-proxy like object into the urql-pipeline 😅 so patch it in `@urql/vue` to be a POJO

EDIT: created a PR to resolve this issue https://github.com/FormidableLabs/urql/pull/1810

**Reply by alexjharrison** [reply]
Awesome.  Thanks for getting the fix up so quick. :1st_place_medal:

**Reply by JoviDeCroock** [reply]
In your code you talk about `JSON.parse(JSON.strigify(results))` was this meant to be `variables` or `requests` with the former code snipper you allude to? Just making sure if there's a possible other issue with for instance actual operation-results?

**Reply by alexjharrison** [reply]
Yes, meant to say requests.  This only concerns saving the outgoing operation to IDB, not the result that comes back.  I haven't come across a case where the results cause any issues.
